subsys/bluetooth/CMakeLists.txt

```diff
@@ -1,10 +1,20 @@
-add_library(subsys__bluetooth INTERFACE)
+# We are just defining avaiable libaries created in the Bluetooth subsystem in order to provide
+# easy understanding og what is available Mesh is part of host in this example.
 
-target_include_directories(subsys__bluetooth INTERFACE ${CMAKE_CURRENT_SOURCE_DIR})
+add_library(subsys_bluetooth STATIC "")
```

Comment:
This breaks userspace/kernelspace memory separation.

To protect kernelspace and userspace memory regions differently the linker needs to place them in different sections.

Today this is implemented by having CMakeLists.txt iterate over the libraries in ZEPHYR_LIBS_PROPERTY and guess based on their names which one's should be in kernelspace and which ones should be in userspace. Not great, I know.

It is implemented here:
https://github.com/zephyrproject-rtos/zephyr/blob/master/CMakeLists.txt#L526

This breaks when you take subsys_bluetooth out of the global list of Zephyr CMake Libraries.

To do this re-organization we need to first find some other mechanism to split up the kernelspace and userspace memory regions.

Reply:
Yes, this I wasn't aware about, that the list were used for placing libs different mem location.
See further comment below, where I list all places ZEPHYR_LIBS_PROPERTY is used